testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksShallowCloneTableCompatibility.java

```diff
@@ -0,0 +1,100 @@
+
+public class TestDeltaLakeDatabricksShallowCloneTableCompatibility
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testShallowCloneSupport()
+    {
+        String baseTable = "test_dl_base_table_" + randomNameSuffix();
+        String shallowClonedTableV1 = "test_dl_shallow_clone_tableV1_" + randomNameSuffix();
+        String shallowClonedTableV2 = "test_dl_shallow_clone_tableV2_" + randomNameSuffix();
+
+        try {
+            // creating base table
+            String createTable = "CREATE TABLE default." + baseTable + " (a_number INT)" + " USING delta " +
+                    " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + baseTable + "'" +
+                    " TBLPROPERTIES (" + " 'delta.minReaderVersion'='2'," + " 'delta.minWriterVersion'='5')";
+            onDelta().executeQuery(createTable);
+            // inserting single row -> version -> 1
+            onDelta().executeQuery("INSERT INTO default." + baseTable + " VALUES (1)");
+
+            // asserting there are only 1 rows when read from both spark and Trino
+            List<QueryAssert.Row> expectedRows = ImmutableList.of(row(1));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + baseTable))
+                    .containsOnly(expectedRows);
+
+            // insert another row -> version -> 1
+            onDelta().executeQuery("INSERT INTO default." + baseTable + " VALUES (2)");
+
+            // creating a shallow clone of Version V1
+            onDelta().executeQuery("CREATE TABLE default." + shallowClonedTableV1 + " SHALLOW CLONE " + "default." + baseTable + " VERSION AS OF 1");
+
+            // asserting there are only 1 rows when read from
+            // spark as of VERSION 1
+            // AND TRINO on shallow cloned table
+            expectedRows = ImmutableList.of(row(1));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + baseTable + " VERSION AS OF 1"))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + shallowClonedTableV1))
+                    .containsOnly(expectedRows);
+
+            // creating a shallow clone of Version V2
+            onDelta().executeQuery("CREATE TABLE default." + shallowClonedTableV2 + " SHALLOW CLONE " + "default." + baseTable + " VERSION AS OF 2");
+
+            // asserting the rows and count to be 2 across, base table read from both Delta and Trino
+            // and also between V2 and shallowClonedTableV2 table
+            expectedRows = ImmutableList.of(row(1), row(2));
```

Comment:
Let's avoid reusing the variable. You can rename it to `expectedV1Rows` & `expectedV2Rows` or something.

Reply:
@ebyhr indeed, renamed and created new one accordingly